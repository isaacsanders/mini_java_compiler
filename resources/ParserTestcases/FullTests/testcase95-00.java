/*/ This is a test /*/
class Test {
    public static void main(String[] args) {
        int i = 20 + 21;
        while (true)
            System.out.println(i);
    }    
}