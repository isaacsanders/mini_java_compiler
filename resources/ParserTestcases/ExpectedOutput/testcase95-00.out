StmtList ::= 
Type ::= INT
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr PLUS Expr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
LiteralExpr ::= TRUE
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= SYSTEMOUTPRINTLN LPAREN Expr RPAREN SEMI
Stmt ::= WHILE LPAREN Expr RPAREN Stmt
StmtList ::= StmtList Stmt
MainClassDecl ::= ...
ClassDeclList ::= 
Program ::= MainClassDecl ClassDeclList
